File: src/SeatCanvas/swift/SVGSeatStyleConfig.swift
//
//  SVGSeatStyleConfig.swift
//  SeatCanvas
//
//  Created by king on 2026/1/16.
//

import Foundation

struct SVGSeatStyleConfig: SeatStyleConfig {
    let type: SeatStyleType = .svg
    let content: String
    init(content: String) {
        self.content = content
    }

    enum CodingKeys: String, CodingKey {
        case type
        case content
    }
}

extension SVGSeatStyleConfig: Encodable {
    func encode(to encoder: Encoder) throws {
        var container = encoder.container(keyedBy: CodingKeys.self)
        try container.encode(type, forKey: .type)
        try container.encode(content, forKey: .content)
    }
}

File: src/SeatCanvas/core/style/SVGSeatStyleConfig.cpp
//
//  SVGSeatStyleConfig.cpp
//  SeatCanvas
//
//  Created by king on 2026/1/16.
//

#include "SVGSeatStyleConfig.hpp"

namespace kk::renderer {
SVGSeatStyleConfig::SVGSeatStyleConfig(const std::string &content)
    : SeatStyleConfig(SeatStyleType::SVG)
    , content(content) {
}

std::shared_ptr<SVGSeatStyleConfig> SVGSeatStyleConfig::Make(const std::string &content) {
    return std::shared_ptr<SVGSeatStyleConfig>(new SVGSeatStyleConfig(content));
}

bool SVGSeatStyleConfig::isEqual(const SeatStyleConfig &other) const {
    const auto *otherConfig = static_cast<const SVGSeatStyleConfig *>(&other);
    if (!otherConfig) {
        return false;
    }
    return content == otherConfig->content;
}
};  // namespace kk::renderer
